code/modules/mob/living/life.dm

```diff
@@ -118,8 +118,15 @@
 
 //this updates all special effects: knockdown, druggy, stuttering, etc..
 /mob/living/proc/handle_status_effects()
-	if(confused)
+	if (confused)
 		confused = max(0, confused - 1)
+	else if (get_confusion())
```

Comment:
Could you turn this into a trait? Check `TRAIT_KNOCKEDOUT` and `TRAIT_INCAPACITATED` on examples of traits added and removed by status effects.
If you could condense the `if(confused)` check into a `if(HAS_TRAIT(src, TRAIT_CONFUSED))` that'd be ideal.

And the status effect should probably process its decay on its own, you shouldn't have to cycle through them all in `handle_status_effects()` for that.

Reply:
Hmm, this would require changing every single use of confusion now so that multiple confusions aren't processed at the same time depending on method, no?

Reply:
There's several ways to do it.
One, for example, is change the additions or removals of confusion.
For example, `confusion = number` becomes `set_confusion(number)` and `confusion += number` becomes either `set_confusion(confusion + number` or something like `add_confusion(number)`

With that kind of setting you can add and remove the trait, properly based on events. Something like this:
```cpp
/mob/living/proc/set_confused(new_value)
	if(confused == new_value)
		return
	. = confused
	confused = new_value
	if(confused)
		if(!.)
			ADD_TRAIT(...)
	else if(.)
		REMOVE_TRAIT(...)
```
This way you won't have to change how the system works. The old system of confusion handles itself as normal, the status effect handles itself.

Unless if you are planning for a larger rework of how confusion happens, and if so it's still better to do it properly, like this.

Reply:
I can do something that simple, yeah.

Reply:
I'm not sure that the status effect handling itself is a good idea because then there would be a mechanical difference between confusion depending on how its applied.

Basically, right now confusion goes down 1 per tick. If the status effect processes itself, but some stuff still just uses the confused variable, then the confused variable will go down per tick, *and* the confusion status effects will go down 1 tick on their own.

Reply:
I'm not sure that the status effect handling itself is a good idea because then there would be a mechanical difference between confusion depending on how its applied.

Basically, right now confusion goes down 1 per tick. If the status effect processes itself, but some stuff still just uses the confused variable, then the confused variable will go down per tick, and the confusion status effects will go down 1 tick on their own.

Reply:
What is the purpose of the status effect? What does it bring on to the table that differentiates it from the life-reducing regular confusion?

Reply:
That it can be removed instantly, which is done when confusion is cured.

Reply:
so it's more of a `confused-until-cured`, why not a trait?
or is it also a timed event?

Reply:
Because confusion isn't a binary thing. You're not just confused or not confused, it scales.

Reply:
Okay, I can see that.
We have two overlapping sources of confusion, which makes this very confusing.
Either we should have a status effect handling things on its own, or the regular old system. Most likely the status effect, since it's more elegant.

Reply:
The regular old system isn't compatible in an easy manner to the behavior I'm trying to achieve, which is nullifying the entire confusion when the symptom is cured.

I'm fine porting everything to the status effect, I'm just unsure if I want to put that into this PR just for the sake of complexity.

Reply:
Yes, the current system doesn't track sources of confusion.

I'd rather we didn't have two concurrent and overlapping confusion effects, that just makes things messy. Make it zero the confusion amount on symptom end as a patch if you want, in the meantime.

Reply:
I can do zeroing out on cure then change confusion later, yeah.